src/Microsoft.Health.Fhir.Api/Features/Filters/ValidateSearchParameterStateRequestAtrribute.cs

```diff
@@ -0,0 +1,70 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Health.Fhir.Core.Features.Operations.SearchParameterState;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    /// <summary>
+    /// A filter that validates the query string for a SearchParameter $Status search request
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    internal sealed class ValidateSearchParameterStateRequestAtrribute : ActionFilterAttribute
+    {
+        private static Dictionary<string, HashSet<string>> _supportedParams = InitSupportedParams();
+
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+            var queryKeys = context.HttpContext.Request.Query.Keys;
+
+            foreach (string key in queryKeys)
+            {
+                if (!_supportedParams[context.HttpContext.Request.Method].Contains(key.ToLowerInvariant()))
+                {
+                    throw new BadRequestException(string.Format(Core.Resources.UnsupportedSearchParameterStateQueryParameter, key));
+                }
+            }
+        }
+
+        private static Dictionary<string, HashSet<string>> InitSupportedParams()
+        {
+            var getParams = new HashSet<string>()
+            {
+                SearchParameterStateProperties.Code,
+                SearchParameterStateProperties.Url,
+                SearchParameterStateProperties.ResourceType,
+                SearchParameterStateProperties.SearchParameterId,
```

Comment:
I believe this refers to the resource id?  if so, the convention in FHIR is to use "_id".  We do have a constant for that defined in Microsoft.Health.Fhir.Core\Features\KnownQueryParameterNames.cs.

Reply:
As I saw your comment in the controller below about waiting to support id based queries, perhaps we can remove .Id from this list of approved params?  I agree we can wait to implement id based searches until we feel it is necessary.

Reply:
Thanks. I will remove it for now, along with the commented code for the id path.